controllers/replicationdestination.go

```diff
@@ -220,3 +221,121 @@ func (r *VolumeSnapshotRestoreReconciler) SetVSRStatus(log logr.Logger) (bool, e
 	r.Log.Info(fmt.Sprintf("waiting for replicationdestination %s/%s to complete", vsr.Spec.ProtectedNamespace, repDestName))
 	return false, nil
 }
+
+func (r *VolumeSnapshotRestoreReconciler) configureRepDestVolOptions(vsr *volsnapmoverv1alpha1.VolumeSnapshotRestore, capacity *resource.Quantity, cm *corev1.ConfigMap) (*volsyncv1alpha1.ReplicationDestinationVolumeOptions, error) {
+
+	if vsr == nil {
+		return nil, errors.New("nil vsb in configureRepDestVolOptions")
+	}
+
+	if capacity == nil {
+		return nil, errors.New("nil pvc in configureRepDestVolOptions")
+	}
+
+	// we do not want users to change these
+	repDestVolOptions := volsyncv1alpha1.ReplicationDestinationVolumeOptions{
+		CopyMethod:              volsyncv1alpha1.CopyMethodDirect,
+		VolumeSnapshotClassName: &vsr.Spec.VolumeSnapshotMoverBackupref.VolumeSnapshotClassName,
+		Capacity:                capacity,
+	}
+
+	var repDestStorageClass string
+
+	var repDestAccessMode string
+	var repDestAccessModeAM []corev1.PersistentVolumeAccessMode
+
+	if cm != nil && cm.Data != nil {
+
+		for spec := range cm.Data {
+
+			// check for config storageClassName, otherwise use source PVC storageClass
+			if spec == "DestinationStorageClassName" {
+				repDestStorageClass = cm.Data["DestinationStorageClassName"]
+
+				// cm.Data is set but field not set
+			} else {
+				repDestStorageClass = vsr.Spec.VolumeSnapshotMoverBackupref.BackedUpPVCData.StorageClassName
+			}
+
+			// check for config accessMode, otherwise use source PVC accessMode
+			if spec == "DestinationAccessMode" {
+				repDestAccessMode = cm.Data["DestinationAccessMode"]
+				repDestAccessModeAM = []corev1.PersistentVolumeAccessMode{corev1.PersistentVolumeAccessMode(repDestAccessMode)}
+
+				// cm.Data is set but field not set
+			} else {
+				repDestAccessModeAM = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+			}
+		}
+
+		// no configmap data used
+	} else {
+
+		repDestStorageClass = vsr.Spec.VolumeSnapshotMoverBackupref.BackedUpPVCData.StorageClassName
+		repDestAccessModeAM = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+	}
```

Comment:
```suggestion
	// default storageClass
	repDestStorageClass = vsr.Spec.VolumeSnapshotMoverBackupref.BackedUpPVCData.StorageClassName
	// default accessMode
	repDestAccessModeAM = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}

	// overwrite defaults if configMap has data
	if cm != nil && cm.Data != nil {
		for spec := range cm.Data {
			// check for config storageClassName, otherwise use source PVC storageClass
			if spec == "DestinationStorageClassName" {
				repDestStorageClass = cm.Data["DestinationStorageClassName"]
			}

			// check for config accessMode, otherwise use source PVC accessMode
			if spec == "DestinationAccessMode" {
				repDestAccessMode = cm.Data["DestinationAccessMode"]
				repDestAccessModeAM = []corev1.PersistentVolumeAccessMode{corev1.PersistentVolumeAccessMode(repDestAccessMode)}
			}
		}
```

Reply:
resolving in favor of https://github.com/migtools/volume-snapshot-mover/pull/191/files#r1115002755